File: core/model/src/main/java/com/chronos/mobile/core/model/AppState.kt
package com.chronos.mobile.core.model

import androidx.compose.runtime.Immutable

@Immutable
data class TimetableSummary(
    val id: String,
    val name: String,
    val courseCount: Int,
    val createdAt: Long,
    val updatedAt: Long,
)

@Immutable
data class AppState(
    val timetables: List<TimetableSummary> = emptyList(),
    val currentTimetableId: String? = null,
    val wallpaperUri: String? = null,
    val currentTimetable: Timetable? = null,
    val themeMode: ThemeMode = ThemeMode.SYSTEM,
    val useDynamicColor: Boolean = false,
)

File: apps/web/src/lib/models/app-state.ts
import type {
	CapsuleCornerStyle,
	PaletteMode,
	ThemeMode,
	Timetable,
	TimetableLayoutMode
} from '@chronos/core';

export type { CapsuleCornerStyle, PaletteMode, ThemeMode, TimetableLayoutMode };

export interface TimetableSummary {
	id: string;
	name: string;
	courseCount: number;
	createdAt: number;
	updatedAt: number;
}

export interface AppState {
	timetables: TimetableSummary[];
	currentTimetableId: string | null;
	wallpaperUri: string | null;
	currentTimetable: Timetable | null;
	themeMode: ThemeMode;
	timetableLayoutMode: TimetableLayoutMode;
	paletteMode: PaletteMode;
	capsuleCornerStyle: CapsuleCornerStyle;
	hapticFeedbackEnabled: boolean;
}

export function emptyAppState(): AppState {
	return {
		timetables: [],
		currentTimetableId: null,
		wallpaperUri: null,
		currentTimetable: null,
		themeMode: 'auto',
		timetableLayoutMode: 'fixed',
		paletteMode: 'vibrant',
		capsuleCornerStyle: 'rounded',
		hapticFeedbackEnabled: true
	};
}
